pkg/validator/validator_test.go

```diff
@@ -216,6 +221,57 @@ func TestValidateOperationalOptions(t *testing.T) {
 	}
 }
 
+func TestValidateFile(t *testing.T) {
+	tests := map[string]struct {
+		path        string
+		wantError   bool
+		notReadable bool
+		expected    string
+	}{
+		"existing file": {
+			path: "../../testdata/inputs/empty.txt",
+		},
+		"unexisting file": {
+			path:      "../../testdata/inputs/this_doesnt_exist.txt",
+			wantError: true,
+			expected:  "file '../../testdata/inputs/this_doesnt_exist.txt' does not exist",
+		},
+		"directory": {
+			path:      "../../testdata/inputs",
+			wantError: true,
+			expected:  "file '../../testdata/inputs' is a directory",
+		},
+		"not readable": {
+			path:        "../../testdata/inputs/noperms.txt",
+			wantError:   true,
+			expected:    "file '../../testdata/inputs/noperms.txt' is not readable",
+			notReadable: true,
+		},
+	}
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			if test.notReadable {
+				err := os.Chmod(test.path, 0222)
+				if err != nil {
+					t.Fatalf("Failed to change permissions of file %s: %s", test.path, err.Error())
+				}
+			}
```

Comment:
This causes some troubles with git, for example checking out another branch fails:
```console
% git checkout master
error: Your local changes to the following files would be overwritten by checkout:
	testdata/inputs/noperms.txt
Please commit your changes or stash them before you switch branches.
Aborting
```
What do you think about setting these permissions directly on the file, given Git should preserve them? In this way, there is also no need to the set them from the test.